### PR TITLE
DataGrid - The detail view cannot be expanded after switching to the edit mode (T845240)

### PR DESCRIPTION
**Pull request by igormalzew**

### CODE REVIEW

File: js/ui/grid_core/ui.grid_core.editing.js [hunk 1]
```diff
@@ -2911,7 +2911,7 @@ module.exports = {
                         editingController.closeEditCell();
                     }
 
-                    return eventName === startEditAction && allowEditing && editingController.editCell(e.rowIndex, columnIndex) || editingController.isEditRow(e.rowIndex);
+                    return eventName === startEditAction && allowEditing && editingController.editCell(e.rowIndex, columnIndex) || allowEditing && editingController.isEditRow(e.rowIndex);
```
Comment (AntonSermyazhko) [inline js/ui/grid_core/ui.grid_core.editing.js]: Suppose this conditions needs to be refactored

Comment (igormalzew) [reply js/ui/grid_core/ui.grid_core.editing.js]: Simplified the condition

File: js/ui/grid_core/ui.grid_core.editing.js [hunk 2]
```diff
@@ -2911,7 +2911,9 @@ module.exports = {
                         editingController.closeEditCell();
                     }
 
-                    return eventName === startEditAction && allowEditing && editingController.editCell(e.rowIndex, columnIndex) || editingController.isEditRow(e.rowIndex);
+                    if(allowEditing && eventName === startEditAction) {
```
Comment (jtoming830) [inline js/ui/grid_core/ui.grid_core.editing.js:2914]: this could be in a separate variable, but it doesn't matter

**Review by jtoming830 — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js [hunk 3]
```diff
@@ -18167,6 +18167,31 @@ QUnit.test('Pressing symbol keys inside detail grid editor does not change maste
     assert.deepEqual(this.keyboardNavigationController._focusedCellPosition, { rowIndex: 0, columnIndex: 1 }, 'Master grid focusedCellPosition is not changed');
 });
 
+QUnit.test('Should open master detail by click if row is edited in row mode (T845240)', function(assert) {
```
Comment (AntonSermyazhko) [inline testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js:18170]: Need to add tests for the different kind of 'startEditAction'

Comment (igormalzew) [reply testing/tests/DevExpress.ui.widgets.dataGrid/dataGrid.tests.js:18170]: Done

File: js/ui/grid_core/ui.grid_core.editing.js (hunk 2 above)
Comment (igormalzew) [reply js/ui/grid_core/ui.grid_core.editing.js:2914]: How can we name this variable? Seems there's no need to extract the condition in this case.

**Review by AntonSermyazhko — APPROVED** [review]

**Review by jtoming830 — APPROVED** [review]

**Review by Alyar666 — APPROVED** [review]

**Pull request merged**